src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1381,28 +1385,66 @@ xclLoadXclBin(const xclBin *buffer)
   xdp::aie::update_device(this);
   xdp::pl_deadlock::update_device(this);
 
-  // scheduler::init can not be skipped even for same_xclbin as in
-  // multiple process stress tests it fails as below init step is
-  // without a lock with above driver load xclbin step.  New KDS fixes
-  // this by ensuring that scheduler init is done by driver itself
-  // along with icap download in single command call and then below
-  // init step in user space is ignored
-  auto ret = xrt_core::scheduler::init(this, buffer);
   START_DEVICE_PROFILING_CB(this);
 
   return ret;
 }
 
+/*
+ * xclDownloadAxlf()
+ */
+int shim::xclDownloadAxlf(drm_xocl_axlf *axlf_obj)
+{
+  int ret = mDev->ioctl(mUserHandle, DRM_IOCTL_XOCL_READ_AXLF, axlf_obj);
+  if (ret && errno == EAGAIN) {
```

Comment:
For future readability
```suggestion
  if (ret && (errno == EAGAIN)) {
```